### PR TITLE
fix: dependency management

### LINKED ISSUE #935: Error with `collocate_data` with `DataInterpolations`
**Describe the bug 🐞**

Collocation with interpolators throws an error since `DataInterpolations` is not imported.

**Minimal Reproducible Example 👇**

```julia
x = randn(1001) |> cumsum
t = collect(0.0:1000)
t2 = collect(0.0:10:1000)
collocate_data(x, t, t2, CubicSpline) 
```

**Error & Stacktrace ⚠️**

```julia
ERROR: UndefVarError: `DataInterpolations` not defined
Stacktrace:
 [1] collocate_data(::Matrix{Float64}, ::Vector{Float64}, ::Vector{Float64}, ::Type)
   @ DiffEqFlux C:\Users\Cornejo\.julia\packages\DiffEqFlux\TglmB\src\collocation.jl:117
 [2] collocate_data(::Vector{Float64}, ::Vector{Float64}, ::Vector{Float64}, ::Type)
   @ DiffEqFlux C:\Users\Cornejo\.julia\packages\DiffEqFlux\TglmB\src\collocation.jl:106
 [3] top-level scope
   @ REPL[73]:1
```

### PR DESCRIPTION
**Pull request by avik-pal**
- removes dependencies that aren't being used
- loads DataInterpolations (fixes #935)
- removes duplicated tests

### CODE REVIEW

File: test/Project.toml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+[deps]
+Aqua = "4c88cf16-eb10-579e-8560-4a9242c79595"
```
[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply test/Project.toml:2]: Merge them back. Using test/Project.toml has some odd behavior with the resolver.

**Comment by avik-pal** [comment]
Still lets wait for the upstream fixes to land before moving ahead with this

**Comment by ChrisRackauckas** [comment]
Well the upstream fixes are in but 😅. Though I will say I did see this come up when playing around. This is why I asked:

> Did ComponentArrays of Tracker always store as Array{TrackedReal} or did it create TrackedArray with views of TrackedArray on sub parts?

When doing ArrayInterface.restructure for the SciMLStructures compatibility, I noticed that TrackedArrays and ComponentArrays didn't mix well. In particular, a TrackedArray on a ComponentArray had the wrappers the wrong way around, and so the indexing didn't work out. And the other way seemed to not make the ComponentArray of TrackedArray structure, but instead TrackedReals. But I know you had to have tried this with Lux before so it must be a recent regression? So what you're seeing there is precisely `ArrayInterface.restructure(x::ComponentArray, y::SomethingTracked)` going awry, and fixing that behavior is the fix. We don't have a test for that exact case in SciMLSensitivity though, would probably be good to add.

**Comment by avik-pal** [comment]
> So what you're seeing there is precisely ArrayInterface.restructure(x::ComponentArray, y::SomethingTracked) going awry, and fixing that behavior is the fix.

This is not hard to fix, I will do it in ComponentArrays

**Comment by avik-pal** [comment]
Once https://github.com/jonniedie/ComponentArrays.jl/pull/269 lands, tests should be green again!

**Pull request merged**